railties/test/generators/shared_generator_tests.rb

```diff
@@ -362,7 +362,7 @@ def test_generator_when_sprockets_is_not_used
 
   def test_dev_option
     run_generator_using_prerelease [destination_root, "--dev"]
-    rails_path = File.expand_path("../../..", Rails.root)
+    rails_path = Rails::Generators::RAILS_DEV_PATH
```

Comment:
Using constant in specs makes me want to test the constant... looking into it.

could do 
```rb
rails_path = File.expand_path("../../..", __dir__)
```
instead for these specs.

Reply:
`Rails.root` & `destination_root` here represent the generated rails app directory. In these tests, that is `../test/fixtures/tmp`, so you can get to the "gem" root by going up from there.  But in actual usage, the generated app could be anywhere in relation to the local gem. But the right path for the rails gem **is always up the same level from here**. It's confusing, but I think __file__ or __dir__ communicate that a little better. We could even change `destination` for the generator and it should still pass.

Reply:
I see where you're coming from. After a quick search `destination_root` is used for going down, but for going up `__dir__` is used more.

Also found this constant which backs up the use of `__dir__` for going up: https://github.com/rails/rails/blob/c6e3336cfdec5ae9c7fb98b03d0196aa56ab30f9/railties/test/isolation/abstract_unit.rb#L29

(this constant is not accessible from your tests except if the file is required, which takes too long IMO)

Reply:
Yes, I think a similar constant for generator tests makes a lot of sense. I will add something.